cli_classic.c

```diff
@@ -555,6 +561,93 @@ int main(int argc, char *argv[])
 		goto out_shutdown;
 	}
 
+	//int ret = spi_send_command(flash, sizeof(cmd), 0, cmd, NULL);
+
+	if (lock){
+
+		unsigned char cmd[1] = {0x05};
+		unsigned char readarr[1];
+		unsigned char sr1, sr2;
+
+		//BUG?: spi_send_command return value is always 0, even if data is read correctly.
+		//	    Therefore spi_send_command return value is not used.
+
+		msg_ginfo("1. Read SR1 and SR2\n");
+
+		//Read status register 1
+		cmd[0] = 0x05;
+		spi_send_command(fill_flash, sizeof(cmd), sizeof(sr1), cmd, &sr1);
+		msg_ginfo("SR1 = 0x%02x\n", sr1);
+
+		//Read status register 2
+		cmd[0] = 0x35;
+		spi_send_command(fill_flash, sizeof(cmd), sizeof(sr2), cmd, &sr2);
+		msg_ginfo("SR2 = 0x%02x\n", sr2);
+
+		if ((sr1 & 0x80) && (sr2 & 0x01)){
+			msg_ginfo("SPI flash is in One Time Program mode. Can't write to it anymore!\n");
+		}
+
+		msg_ginfo("2. Write new values to SR1 = 0x34, SR2 = 0x40\n");
+
+		// Write enable command
+		cmd[0] = 0x06;
+		spi_send_command(fill_flash, sizeof(cmd), 0, cmd, NULL);
+
+		// Write status register 1
+		sr1 &= 0x83;		// 
+		sr1 |= 0x34;		// Set SEC, TB, BP2, BP1, BP0 bits
+		sr2 &= 0x41;
+		sr2 |= 0x40;		// Enable CMP to reach WP_RO region (0x200000 - 0x77777)
+		unsigned char cmd_write[3] = {0x01, sr1, sr2};
+		spi_send_command(fill_flash, sizeof(cmd_write), 0, cmd_write, NULL);
+
+		msg_ginfo("3. Read updated SR1 i SR2\n");
+		//Verify if SR1 and SR2 are written correctly
+		// Read status register 1
+		cmd[0] = 0x05;
+		spi_send_command(fill_flash, sizeof(cmd), sizeof(readarr), cmd, readarr);
+		msg_ginfo("SR1 = 0x%02x\n", readarr[0]);
+
+		// Read status register 2
+		cmd[0] = 0x35;
+		spi_send_command(fill_flash, sizeof(cmd), sizeof(readarr), cmd, readarr);
+		msg_ginfo("SR2 = 0x%02x\n", readarr[0]);
+
+		/*
+		// uncomment when read/write commands will be verified as 100% working
+		// Set SPI flash in One Time Program mode
+
+		msg_ginfo("4. Set OTP mode. First, Write enable.\n");
+		// Write enable command
+		cmd[0] = 0x06;
+		spi_send_command(fill_flash, sizeof(cmd), 0, cmd, NULL);
+
+		msg_ginfo("5. SR1.7 = 1 and SR2.0 = 1\n");    // SRP = 1 and SRP1 = 1 means OTP mode
+		// Write status register 1
+		sr1 &= 0x7F;		// 
```

Comment:
These logical & is useless here. First you clear SRP0 to set it again in next line @piotr-kleins . Update the masks as they are in sortbootorder (mask only writable bits)